### PR TITLE
replace spaces with underscores in screen routes

### LINKED ISSUE #2146: Space in url breaks the route
**Describe the bug**
It is possible to add a space in a route input field (when creating a new screen or when updating the route of an existing one). This results in a broken route

**To Reproduce**
1. Create a new screen
2. Enter '/broken url' in the url field 
3. Save the screen
4. Start the app
5. Click the link in the header
6. Note: the page is not loaded

**Expected behavior**
In order to prevent this kind of behavior we should either replace spaces with an underscore ('_') or prevent the user from inserting a space (ignoring the space key). 

**Screenshots**
Not applicable 

**Desktop (please complete the following information):**
 - OS: MacOS 11.5
 - Browser Vivaldi
 - Version Develop branch

### PR DESCRIPTION
**Pull request by mslourens**
## Description
Fixes #2146: replace spaces in screen route input fields with underscores to prevent breaking routes

### CODE REVIEW

**Comment by mslourens** [comment]
the build fails for no particular reason, locally all Cypress tests succeed. I am not allowed to re-run the tests, so if one of you guys could do that, that would be awesome.

**Comment by Rory-Powell** [comment]
@mslourens kicked them off again

**Review by shogunpurple — CHANGES_REQUESTED** [review]
Hi @mslourens thanks again for the contribution. I had a chat with the team and right now we use dashes for auto screen routes generated from tables with a space. If we change from underscores to dashes in this PR to keep things consistent, I'll get it merged!

**Comment by shogunpurple** [comment]
Also @mslourens I would just like to say a huge thank you from the team for your contributions to the project.

You have provided an immense amount of value in such a short time! We would like to invite you to our discord channel, where we have a chat for contributors.

https://discord.gg/xSQnaypz

We would also love to hear more about your use case with budibase as well understand any way we can help you out.

File: packages/builder/cypress/integration/createScreen.js [hunk 1]
```diff
@@ -7,5 +7,15 @@ context("Screen Tests", () => {
...
     cy.createScreen("Test Screen", "/test")
+    cy.get(".nav-items-container").within(() => {
+      cy.contains("/test").should("exist")
+    })
+  })
+
+  it("Should update the url", () => {
+    cy.createScreen("Test Screen", "test with spaces")
+    cy.get(".nav-items-container").within(() => {
+      cy.contains("/test_with_spaces").should("exist")
```
Comment (shogunpurple) [inline packages/builder/cypress/integration/createScreen.js:18]: @mslourens looks like the cypress tests are failing due to the switch to dashes. Update this to:

```
test-with-spaces
```

and we are good to go.

**Pull request merged**